File: group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        sorted_strs = [''.join(sorted(string)) for string in strs]
        grouped_strs = []

        hashmap = {}
        for i in range(len(strs)):
            sorted_str = sorted_strs[i]
            if sorted_str not in hashmap:
                hashmap[sorted_str] = len(grouped_strs)
                grouped_strs.append([])
            grouped_strs[hashmap[sorted_str]].append(strs[i])
        
        return grouped_strs

File: c++/group-anagrams.cc
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::vector<std::vector<std::string>> anagramGroups;
        std::unordered_map<std::string, std::vector<std::string>> sorted_anagrams;
        for (const auto& str : strs)
        {
            std::string sorted_str = str;
            sort(sorted_str.begin(), sorted_str.end());
            sorted_anagrams[sorted_str].push_back(str);
        }
        for (const auto& [key, val] : sorted_anagrams)
        {
            anagramGroups.push_back(val);
        }
        return anagramGroups;
    }                                                                                                       
};
